src/Options.php

```diff
@@ -558,7 +577,40 @@ public function setIntegrations(array $integrations): void
      */
     public function getIntegrations(): array
     {
-        return $this->options['integrations'];
+        $defaultIntegrations = $this->options['default_integrations'] ? $this->defaultIntegrations : [];
+        $userIntegrations = $this->options['integrations'];
+        $integrations = [];
+
+        if (\is_array($userIntegrations)) {
```

Comment:
I would reverse the check, and do a 
```suggestion
        if (\is_callable($userIntegrations)) {
            return $userIntegrations($defaultIntegrations);
        }
```
to reduce the indentation level of the whole method.